release-tools/prow.sh

```diff
@@ -865,6 +875,133 @@ install_snapshot_controller() {
   done
 }
 
+# Installs all necessary volumepopulators CRDs
+install_volumepopulators_crds() {
+  # Wait until volumepopulators CRDs are in place.
+  CRD_BASE_DIR="https://raw.githubusercontent.com/kubernetes-csi/volume-data-source-validator/${VOLUME_DATA_SOURCE_VALIDATOR_VERSION}/client/config/crd"
+  if [[ ${REPO_DIR} == *"volume-data-source-validator"* ]]; then
+      CRD_BASE_DIR="${REPO_DIR}/client/config/crd"
+  fi
+  echo "Installing volumepopulators CRDs from ${CRD_BASE_DIR}"
+  kubectl apply -f "${CRD_BASE_DIR}/populator.storage.k8s.io_volumepopulators.yaml" --validate=false
+  cnt=0
+  until kubectl get volumepopulators.populator.storage.k8s.io; do
+    if [ $cnt -gt 30 ]; then
+        echo >&2 "ERROR: volumepopulators CRDs not ready after over 1 min"
+        exit 1
+    fi
+    echo "$(date +%H:%M:%S)" "waiting for volumepopulators CRDs, attempt #$cnt"
+	cnt=$((cnt + 1))
+    sleep 2
+  done
+}
+
+# Install snapshot controller and associated RBAC, retrying until the pod is running.
+install_volume_data_source_validator_controller() {
+  CONTROLLER_DIR="https://raw.githubusercontent.com/kubernetes-csi/volume-data-source-validator/${VOLUME_DATA_SOURCE_VALIDATOR_VERSION}"
+  if [[ ${REPO_DIR} == *"volume-data-source-validator"* ]]; then
+      CONTROLLER_DIR="${REPO_DIR}"
+  fi
+  VOLUME_DATA_SOURCE_VALIDATOR_RBAC_YAML="${CONTROLLER_DIR}/deploy/kubernetes/rbac-data-source-validator.yaml"
+  echo "kubectl apply -f ${SNAPSHOT_RBAC_YAML}"
+  # Ignore: Double quote to prevent globbing and word splitting.
+  # shellcheck disable=SC2086
+  kubectl apply -f ${SNAPSHOT_RBAC_YAML}
+
+  cnt=0
+  until kubectl get clusterrolebinding volume-data-source-validator; do
+     if [ $cnt -gt 30 ]; then
+        echo "Cluster role bindings:"
+        kubectl describe clusterrolebinding
+        echo >&2 "ERROR: volume data source validator controller RBAC not ready after over 5 min"
+        exit 1
+    fi
+    echo "$(date +%H:%M:%S)" "waiting for snapshot RBAC setup complete, attempt #$cnt"
+	cnt=$((cnt + 1))
+    sleep 10
+  done
+
+  VOLUME_DATA_SOURCE_VALIDATOR_CONTROLLER_YAML="${CONTROLLER_DIR}/deploy/kubernetes/setup-data-source-validator.yaml"
+  if [[ ${REPO_DIR} == *"volume-data-source-validator"* ]]; then
+      # volume-data-source-validator-controller image built from the PR will get a "csiprow" tag.
+      # Load it into the "kind" cluster so that we can deploy it.
+      NEW_TAG="csiprow"
+      NEW_IMG="volume-data-source-validator-controller:${NEW_TAG}"
+      echo "kind load docker-image --name csi-prow ${NEW_IMG}"
+      kind load docker-image --name csi-prow ${NEW_IMG} || die "could not load the volume-data-source-validator:csiprow image into the kind cluster"
```

Comment:
Instead of `kind load` the image as "volume-data-source-validator:csiprow" here, I think that we can `kind load` the image as "registry.k8s.io/sig-storage/volume-data-source-validator:v1.0.0" as defined in https://github.com/kubernetes/kubernetes/blob/master/test/e2e/testing-manifests/storage-csi/any-volume-datasource/volume-data-source-validator/setup-data-source-validator.yaml#L20 . Then, the image used in the e2e tests will be the one built in the CI job.


The other way will be to modify the e2e codes around https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/provisioning.go#L268-L276 to allow modifying the image used to be "volume-data-source-validator:csiprow". However, I'm not sure if there is an easy way to achieve it (We may be able to conditionally patch the manifest for this purpose, but we need to consider how to check the condition).

Reply:
Sounds good.

@bswartz Could you add a specific tag to the image that should be used in the e2e test in k/k?